doc-content/jbpm-docs/src/main/asciidoc/Integration/spring-process-services-config-proc.adoc

```diff
@@ -0,0 +1,126 @@
+[id='spring-process-services-config-proc_{CONTEXT}']
+= Configuring {PROCESS_ENGINE} services with a Spring application
+
+The following procedure is an example of complete configuration for {PROCESS_ENGINE} services within a Spring application.
+
+.Procedure
+
+. Configure transactons:
++
+.Configuring transactions in the `spring.xml` file
+[source,xml]
+----
+<context:annotation-config />
+<tx:annotation-driven />
+<tx:jta-transaction-manager />
+
+<bean id="transactionManager" class="org.springframework.transaction.jta.JtaTransactionManager" />
+----
++
+. Configure JPA and persistence:
++
+.Configuring JPA and persistence in the `spring.xml` file
+[source,xml]
+----
+<bean id="entityManagerFactory" class="org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean" depends-on="transactionManager">
+   <property name="persistenceXmlLocation" value="classpath:/META-INF/jbpm-persistence.xml" />
+</bean>
+----
++
+. Configure security and user and group information providers:
++
+.Configuring security and user and group information providers in the `spring.xml` file
+[source,xml]
+----
+<util:properties id="roleProperties" location="classpath:/roles.properties" />
+
+<bean id="userGroupCallback" class="org.jbpm.services.task.identity.JBossUserGroupCallbackImpl">
+  <constructor-arg name="userGroups" ref="roleProperties"></constructor-arg>
+</bean>
+
+<bean id="identityProvider" class="org.jbpm.spring.SpringSecurityIdentityProvider"/>
+----
++
+. Configure the runtime manager factory. This factory is Spring context aware, so it can interact with the Spring container in correct way and support the necessary services, including the transactional command service and the task service:
++
+.Configuring the runtime manager factory in the `spring.xml` file
+[source,xml]
+----
+<bean id="runtimeManagerFactory" class="org.kie.spring.manager.SpringRuntimeManagerFactoryImpl">
+  <property name="transactionManager" ref="transactionManager"/>
+  <property name="userGroupCallback" ref="userGroupCallback"/>
+</bean>
+
+<bean id="transactionCmdService" class="org.jbpm.shared.services.impl.TransactionalCommandService">
+  <constructor-arg name="emf" ref="entityManagerFactory"></constructor-arg>
+</bean>
+
+<bean id="taskService" class="org.kie.spring.factorybeans.TaskServiceFactoryBean" destroy-method="close">
+  <property name="entityManagerFactory" ref="entityManagerFactory"/>
+  <property name="transactionManager" ref="transactionManager"/>
+  <property name="userGroupCallback" ref="userGroupCallback"/>
+  <property name="listeners">
+    <list>
+      <bean class="org.jbpm.services.task.audit.JPATaskLifeCycleEventListener">
+        <constructor-arg value="true"/>
+      </bean>
+    </list>
+  </property>
+</bean>
+----
++
+. Configure {PROCESS_ENGINE} services as Spring beans:
++
+.Configuring {PROCESS_ENGINE} services as Spring beans in the `spring.xml` file
+[source,xml]
+----
+<!-- definition service -->
```

Comment:
I mentioned in the final polish of the last process engine PR, but as a nice-to-have, would be ideal for the snippet comments to be consistent and compliant with capitalization, so "Definition service", "Runtime data service ..", "Register runtime data service ...". Not as big of a deal with just noun-phrase comments, but with commands or sentences like the last one I mentioned, just feels rough and untidy when capitalization is inconsistent and/or incorrect. Just a thought for snippets throughout and in general. Something I try to do myself, even when it's exactly as it appears in the code source. Still best to make at least better in our professional docs. Not critical.